src/types.ts

```diff
@@ -113,6 +117,22 @@ export interface PredefinedType extends BaseValueType {
   name: string;
 }
 
+export interface LiteralType extends BaseValueType {
+  kind: ValueTypeKind.literalType;
+  type:
+    | {
+        kind: ValueTypeKind.basicType;
+        value: BasicTypeValue.string | BasicTypeValue.number | BasicTypeValue.boolean;
+      }
+    | EnumType;
+  value: Value;
+}
+
+export interface UnionType extends BaseValueType {
```

Comment:
How about aligning the definition of `UnionType` with `EnumType` like this:

```typescript
export interface UnionType extends BaseValueType {
  kind: ValueTypeKind.unionType;
  subType: EnumSubType;
  members: Value[];
}
```

Reply:
I can rename members, subtypes are already included in the LiteralType.

Reply:
If all members should only be in a single subtype, doesn't it make more sense to not include type info in every member?

Reply:
Then maybe it is better to create a EnumType object directly. No need to create a union and convert it to enum later

Reply:
Wait, I still need a union type so that I can override its name.

Reply:
Since typescript's union supports different types, let's keep this UnionType as well.

Reply:
The types in this file are not designed to mimic TypeScript types, but defined as types ts-gyb is capable of handling. They are more aligned with destination languages Swift and Kotlin. It would be better to make sure all the values of these type definitions are valid values that renders can safely generate code with.

Reply:
Even through, Swift/Kotlin doesn't have union type (yet), so keeping the original type is better for now.

Reply:
I thought it would be better to keep it consistent, and since tuple type is defined as a nameless interface type here, it would be intuitive to define the union type as a nameless enum type. However, it's fine by me if you want to keep the `UnionType` definition in its current state.